### ISSUE TITLE
Aircraft .json format

### ISSUE THREAD

**Issue by gvanem** [alkissack/Dump1090-OpenLayers3-html#7]
Hello.

I've been unsuccessful trying to use this web-root in my [fork](https://github.com/gvanem/Dump1090/) of the original Dump1090 by Salvatore Sanfilippo.

When I handle the `GET /data/aircraft.json`, I send back to the web-client an array like this:
```
[
  {"hex":"471234", "flight":"abcd", "lat":60.1, "lon":-5.0, "altitude":4000, "track":140, "speed":300},
  ...
]
```

But nothing gets shown on the nice map.
What other `.json` array does it need to function? Is this documented somewhere?

Please advice.

**Comment by alkissack** [comment]
Can you let me know what you are trying to achieve. Are you using some code from this branch and/or what is the problem?

**Comment by gvanem** [comment]
> Can you let me know what you are trying to achieve. 

I'm simply trying to make your `public_html/index.html` + all the `.js` files work with my version of Dump1090.

For the client to work, the server should send it's Json data on a format the server expects. I fail to understand what format that it. Looking at how the Dump1090-FA formats it's `/data/aircraft.json`, I assume it's done in [`generateAircraftJson()`](https://github.com/flightaware/dump1090/blob/master/net_io.c#L1607). No?

I've been using the *Chrome Developer Console* to see which exceptions triggers in `script.js`. For example:
```js
function processReceiverUpdate(data) {
	// Loop through all the planes in the data packet
        var now = data.now;
        var acs = data.aircraft;
  ....
	for (var j=0; j < acs.length; j++) {
                var ac = acs[j];
                var hex = ac.hex;
                var squawk = ac.squawk;
                var plane = null;
```

What is `acs.length`? I know almost nothing of JavaScript. So I fail to understand the relation between JS-code and JSON-data.
Should it be something like:
```
[
  ...
   "aircraft": [{"hex":"47809F", "flight":"WIF08J  ", "lat":60.405858, "lon":5.193296, "altitude":3750, "track":0, "speed":160}], 
    "length":1}
]
```

> Are you using some code from this branch

Yes. I use all your `public_html` files as my web-root in my Dump1090 program.

**Comment by alkissack** [comment]
The full and current aircraft.json can be seen at : <your web address for dump1090>/data/aircraft.json
I pesume this has the aircraft in it, when you run it?  Can you email me the output (2021 @ kissack . co . uk)
acs.lenght is the length of the json in aircraft objects. Your second code example (json) does look to be differnt to mine
`"aircraft" : [
    {"hex":"398477","alt_baro":36000,"gs":448.0,"tas":442,"track":354.6,"roll":-15.1,"squawk":"2334","emergency":"none","version":0,"mlat":[],"tisb":[],"messages":40,"seen":0.1,"rssi":-15.2},
 `
what version of dump1090-fa are you using?

**Comment by gvanem** [comment]
> what version of dump1090-fa are you using?

None. Read my posts again. I wrote "...*my fork*" and "*... with my version of Dump1090*."

After I filed this issue, I've managed to compile Dump1090-FA on Windows (MSVC/clang).
Since the `--write-json` option did nothing for `_WIN32`, no `<dir>/data/aircraft.json` file was generated.

I fixed that and now I see how the format is supposed to be.  So I can figure out the .JSON format from there and
close this issue.
